packages/effect/src/internal/stream.ts

```diff
@@ -3133,6 +3135,31 @@ export const fromPubSub: {
   return options?.shutdown ? ensuring(stream, PubSub.shutdown(pubsub)) : stream
 }
 
+/** @internal */
+export const fromTPubSub: {
+  <A>(pubsub: TPubSub.TPubSub<A>, options: {
+    readonly scoped: true
+    readonly shutdown?: boolean | undefined
+  }): Effect.Effect<Stream.Stream<A>, never, Scope.Scope>
+  <A>(pubsub: TPubSub.TPubSub<A>, options?: {
+    readonly scoped?: false | undefined
+    readonly shutdown?: boolean | undefined
+  }): Stream.Stream<A>
+} = (pubsub, options): any => {
+  if (options?.scoped) {
+    const effect = Effect.map(
+      TPubSub.subscribeScoped(pubsub),
+      (queue) => fromTQueue(queue, { shutdown: true })
```

Comment:
This seems strange to me, as the `Scope` should be responsible for shutting down the subscription.

Reply:
good catch, the full options story didn't quite make sense, cleared up in https://github.com/Effect-TS/effect/pull/2725/commits/55dcb28e25e473eb172d20d43b817f4c0aa7d7f4